main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"log"
+	"os"
+	"image"
+	"image/jpeg"
+	"image/color"
+	"yi/point"
+	"yi/harris"
+)
+
+func markPoint(p point.Point, img *image.RGBA64) {
+	red := color.RGBA{
+		R: 1 << 8 - 1,
+		G: 0,
+		B: 0,
+		A: 1 << 8 - 1,
+	}
+	for i := p.X - 10; i <= p.X + 10; i++ {
+		img.Set(i, p.Y, red)
+		img.Set(i, p.Y + 1, red)
+	}
+	for j := p.Y - 10; j <= p.Y + 10; j++ {
+		img.Set(p.X, j, red)
+		img.Set(p.X + 1, j, red)
+	}
+}
+
+func main() {
+	log.SetOutput(os.Stdout)
+	file, err := os.Open("/Users/SU/projects/gillnet/go/src/yi/test-3.jpg")
+	if err != nil {
+		log.Fatal("Error while opening file")
+	}
+	defer file.Close()
+
+	img, _, err := image.Decode(file)
+	if err != nil {
+		log.Fatal("error decoding image")
+	}
+
+	newRec := img.Bounds()
+	newImg := image.NewRGBA64(img.Bounds())
+	greyImg := image.NewGray(img.Bounds())
+
+	for i := newRec.Min.X; i <= newRec.Max.X; i++ {
+		for j := newRec.Min.Y; j <= newRec.Max.Y; j++ {
+			newImg.Set(i, j, img.At(i, j))
+			greyImg.Set(i, j, img.At(i, j))
+		}
+	}
```

Comment:
move some of them to it's own package 	28600f9, it's still a surprise to me the library does not provide a way to transfer image color more easily.